libbpf-tools/tcpconnlat.bpf.c

```diff
@@ -85,12 +72,12 @@ int BPF_PROG(tcp_rcv_state_process, struct sock *sk)
 			sizeof(event.comm));
 	event.ts_us = ts / 1000;
 	event.tgid = piddatap->tgid;
-	event.lport = sk->__sk_common.skc_num;
-	event.dport = sk->__sk_common.skc_dport;
-	event.af = sk->__sk_common.skc_family;
+	event.lport = BPF_CORE_READ(sk, __sk_common.skc_num);
```

Comment:
I wonder any specific reason for choosing BPF_CORE_READ?  Cuz i think  BTF-enabled prog should be given top prio in terms of performance.  does kernel 5.4 not support btf relocation for this or sth else?